File: samples/sample_plugin.py
"""
sample_plugin.py - Plugin ultra avancé pour Threed (Python)
Inclut : hooks, audit, sécurité, CI/CD, doc intégrée.
"""

class SamplePlugin:
    def __init__(self):
        self.name = 'SamplePlugin'
        self.version = '2.0.0'
        self.enabled = False
        self.audit_trail = []

    def activate(self, context=None):
        self.enabled = True
        self.audit_trail.append({'date': __import__('datetime').datetime.now().isoformat(), 'action': 'activated', 'context': context})
        # Sécurité : vérifier le contexte d’activation
        if context and context.get('user', {}).get('role') != 'admin':
            raise Exception('Activation non autorisée')
        print(f'[PLUGIN][{self.name}] Activé.')

    def deactivate(self, context=None):
        self.enabled = False
        self.audit_trail.append({'date': __import__('datetime').datetime.now().isoformat(), 'action': 'deactivated', 'context': context})
        print(f'[PLUGIN][{self.name}] Désactivé.')

    def run(self, data):
        if not self.enabled:
            raise Exception('Plugin désactivé')
        self.audit_trail.append({'date': __import__('datetime').datetime.now().isoformat(), 'action': 'run', 'data': data})
        return f'Traitement avancé Threed: {data}'

    def get_audit_trail(self):
        return self.audit_trail

File: samples/sample_plugin.js
// sample_plugin.js – Plugin ultra avancé pour Threed (Node.js)
/**
 * Exemple de plugin avancé pour Threed (3D)
 */
class SampleThreedPlugin {
  constructor() {
    this.name = 'SampleThreedPlugin';
  }
  run(data) {
    // Traitement avancé sur les données 3D
    return { ...data, plugin: this.name, status: 'processed' };
  }
}

module.exports = SampleThreedPlugin;

/**
 * Documentation intégrée :
 * - Support hooks, audit, sécurité, CI/CD
 * - Utilisable pour tests unitaires et intégration
 */
